db/Pos/DB/BlockIndex.hs

```diff
@@ -0,0 +1,36 @@
+-- | Operations with block index db.
+
+module Pos.DB.BlockIndex
```

Comment:
I don't get it. Why would you create a module that provides only read capabilities over block index?

AFAIK every other db module follows the following hierarchy:
1. Reader methods
2. Writer batch functions
3. Iterators
4. Prefix keys

It seems I am missing something important. What exactly?

Update: I now seem to have more understanding on this question (after i've read through all the PR), but I would like you to explain anyway. Like, yes -- we always write blunds only, but design is controversial.

Reply:
1. There are only readers methods.
2. We don't write a batch to block index db, just write some information along with putting a block to block db.
3. We don't iterate over headers, we use BlockExtra for it which is stored into GState-db
4. it's done

:paket:

Reply:
Sorry, but it doesn't really answer my question.
I suggest to export a function instead ofa  prefix key. I find current solution unsafe (everybody from outside can write to `blockIndexKey` something which is different from `HeaderHash`.

Reply:
breten, vso horosho?
`blockIndexKey :: HeaderHash -> ByteString`

Reply:
It's not an interface we're providing -- that's what I'm trying to say. We don't usually export index keys.
Yeah, whatever, I guess you won't ever understand :pes: :peka: